selfhost/main.jakt

```diff
@@ -472,65 +573,166 @@ function main(args: [String]) {
         return 0
     }
 
-    let output = CodeGenerator::generate(compiler, checked_program, debug_info: codegen_debug)
+    let codegen_result = CodeGenerator::generate(compiler, checked_program, debug_info: codegen_debug)
 
-    mut cpp_filename: String = ""
-    mut output_filename: String = ""
+    mut depfile_builder = StringBuilder::create()
 
-    if (write_source_to_file or build_executable or run_executable) {
-        
-        if set_output_filename.has_value() {
-            cpp_filename = binary_dir + "/" + set_output_filename! + ".cpp"
-            output_filename = binary_dir + "/" + set_output_filename!
```

Comment:
Custom output filename isn't being used currently.

Reply:
It's a meaningless flag now, will remove.

Reply:
If it is removed how would we give the binary a custom name now?

Reply:
yep, used it for the executable file name.